Add extractor subroutines to CRS

schrauf opened the pull request:
Dear,

I made some modifications to use the libsparse module:

Added extractor subroutines for CRS objects which allow to extract the row pointers, the column indices and non-zero values:
call A%rowptr(iA)
call A%colval(jA)
call A%nzval(vA)
These subroutines check that the array provided has the same length as the corresponding internal component of the CRS object.

Also added an optional msglvl parameter for the solver subroutines to pass to pardiso.

Lastly, .gitmodule was modified to use HTTPS (so that it can be used without setting up an ssh key in github). If this last modification is an issue, I can remove it and look at the alternatives.

jvdp1 commented on src/modsparse_crs.f90:
```diff
@@ -209,6 +209,46 @@ module subroutine external_crs(sparse,ia,ja,a)
+ 
+  if(size(ia).ne.size(sparse%ia))then
+   write(sparse%unlog,'(a)')' ERROR: The provided array ia is of a different size!'
+   stop
```
```suggestion
   error stop
```

jvdp1 commented on src/modsparse_crs.f90:
```diff
@@ -209,6 +209,46 @@ module subroutine external_crs(sparse,ia,ja,a)
+ 
+  if(size(ja).ne.size(sparse%ja))then
+   write(sparse%unlog,'(a)')' ERROR: The provided array ja is of a different size!'
+   stop
```
```suggestion
   error stop
```

jvdp1 commented on src/modsparse_crs.f90:
```diff
@@ -209,6 +209,46 @@ module subroutine external_crs(sparse,ia,ja,a)
+ 
+  if(size(a).ne.size(sparse%a))then
+   write(sparse%unlog,'(a)')' ERROR: The provided array a is of a different size!'
+   stop
```
```suggestion
   error stop
```

jvdp1 reviewed (approved):
thank you. LGTM.
Could you add a few tests for the 3 get_ subroutines please?

schrauf commented:
Hi,
added a test function for the three new get getters.
I thought it was good to test them together with the "external" subroutine, which was not covered.

(Also modified the getter subroutines to allocate the output array within each subroutine)

jvdp1 commented:
Thank you @schrauf . I'll merge it.
